Separated out feature checking from selftest

radarhere opened the pull request:
#1182 suggests separating out feature checking code from selftest.py into it's own module, so that users can check the availability of features at any point.

This is a basic attempt at that idea.

homm commented on PIL/features.py:
```diff
@@ -0,0 +1,71 @@
+
+
+def get_all_modules():
+    # While the dictionary keys could be used here,
```
It is better to use list of tuples instead of duplication. `get_all_modules` will be unnecessary:

```python
modules = [
    ("PIL CORE", "PIL._imaging"),
    ("TKINTER", "PIL._imagingtk"),
    ...
]

def get_supported_modules():
    supported_modules = []
    for name, module in modules:
        if check_module(module):
            supported_modules.append(name)
    return supported_modules


def check_module(module):
    method_to_call = None

```

radarhere replied:
I'll make a commit shortly using the tuples idea, but my plan was that people would be able to call

    check_module("TKINTER")

homm replied:
But it was labels, not identifiers. I think it is not very convenient to call `check_module('Transparent WEBP')` or `check_codec('ZLIB (PNG/ZIP)')`. Maybe we should define some constants, for example 'transp_webp' and 'png'.

```python
modules = [
    ...
    ('webp', "WEBP", "PIL._webp"),
    ('transp_webp', "Transparent WEBP", "WEBP", "WebPDecoderBuggyAlpha"),
]
```

hugovk commented:
Tests for features.py please, so we can be sure it runs fine on all supported platforms. 

Of course, there's no point asserting that any particular features are present, as we have no idea if [Travis Linux/Travis Mac/mythical Windows CI/someone's own machine] has those, but just to check it's all callable.

radarhere commented:
Okay, added tests. I also found that setup referred to 'OPENJPEG (JPEG2000)' while selftest referred to 'JPEG 2000', so I changed that to be the same.

hugovk commented:
Thanks.

mgedmin commented:
Thank you!

(It would be even nicer if the API for this were documented.)

aclark4life commented:
Nice

hugovk commented:
Docs would be good. In the meantime, it's sort-of documented by the tests: https://github.com/python-pillow/Pillow/blob/master/Tests/test_features.py
